### PR TITLE
Strict mode integration tests

### PR DESCRIPTION
**Pull request by JojiiOfficial**
Adds integration tests for strict mode

### CODE REVIEW

File: tests/openapi/test_strictmode.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+                "limit": 3
+            }
+        )
+
+    search_request_with_timeout(3).raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "max_timeout": 2,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:145]:
```suggestion
    set_strict_mode({
        "enabled": True,
        "max_timeout": 2,
    })
    
    search_request_with_timeout(2).raise_for_status()
```

File: tests/openapi/test_strictmode.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+                },
+            }
+        )
+
+    search_request_with_filter().raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "unindexed_filtering_retrieve": False,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:181]:
```suggestion
    set_strict_mode({
        "enabled": True,
        "unindexed_filtering_retrieve": True,
    })
    
    search_request_with_filter().raise_for_status()
    
    set_strict_mode({
        "unindexed_filtering_retrieve": False,
    })
```

File: tests/openapi/test_strictmode.py [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+    update_request_with_filter().raise_for_status()
+
+    # Reset any changes
+    basic_collection_setup(collection_name=collection_name)
+
+    set_strict_mode({
+        "enabled": True,
+        "unindexed_filtering_update": False,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:233]:
```suggestion
    set_strict_mode({
        "enabled": True,
        "unindexed_filtering_update": True,
    })
    
    update_request_with_filter().raise_for_status()
    
    set_strict_mode({
        "unindexed_filtering_update": False,
    })
```

File: tests/openapi/test_strictmode.py [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+                }
+            }
+        )
+
+    search_request().raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "search_max_hnsw_ef": 3,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:278]: Note: here we set it to 3 but I think 4 should already fail (untested):

```suggestion
    set_strict_mode({
        "enabled": True,
        "search_max_hnsw_ef": 5,
    })
    
    search_request().raise_for_status()
    
    set_strict_mode({
        "search_max_hnsw_ef": 4,
    })
```

File: tests/openapi/test_strictmode.py [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+                }
+            }
+        )
+
+    search_request().raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "search_allow_exact": False,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:307]:
```suggestion
    set_strict_mode({
        "enabled": True,
        "search_allow_exact": True,
    })
    
    search_request().raise_for_status()
    
    set_strict_mode({
        "search_allow_exact": False,
    })
```

File: tests/openapi/test_strictmode.py [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+                }
+            }
+        )
+
+    search_request().raise_for_status()
+
+    set_strict_mode({
+        "enabled": True,
+        "search_max_oversampling": 1.0,
+    })
```
Comment (timvisee) [inline tests/openapi/test_strictmode.py:337]:
```suggestion
    set_strict_mode({
        "enabled": True,
        "search_max_oversampling": 2.0,
    })
    
    search_request().raise_for_status()
    
    set_strict_mode({
        "enabled": True,
        "search_max_oversampling": 1.9,
    })
```

**Review by timvisee — COMMENTED** [review]
Even though it may be a bit excessive, let's also test it still works if a higher limit is configured while we're at it.

I believe I've covered them in the review suggestions below:

File: tests/openapi/test_strictmode.py [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+    
+    set_strict_mode({
+        "unindexed_filtering_retrieve": False,
+    })
+
+    search_fail = search_request_with_filter()
+
+    assert search_fail.json()['status']['error'] == ('Forbidden: Index required but not found for "city" of one of the '
+                                                     'following types: [keyword]. Help: Create an index for this key '
+                                                     'or use a different filter.')
```
[the remark this answers is not in the document]
Comment (JojiiOfficial) [reply tests/openapi/test_strictmode.py:201]: Do you have a suggestion for a better approach? Only check whether it's an error or not?

**Review by timvisee — APPROVED** [review]

File: tests/openapi/test_strictmode.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (timvisee) [reply tests/openapi/test_strictmode.py:201]: We all agreed to simplify this just to: check the field name is in the error message

**Review by generall — APPROVED** [review]

**Pull request merged**